tooling/noir_js_backend_barretenberg/src/index.ts

```diff
@@ -73,6 +62,7 @@ export class BarretenbergBackend implements Backend {
   }
 
   async generateProof(decompressedWitness: Uint8Array, makeEasyToVerifyInCircuit: boolean): Promise<Uint8Array> {
+    await this.instantiate();
```

Comment:
Just a smaller nit...

as suggested in slack, I wouldn't `instantiante` these dynamically, as we're adding some overhead to every proof generation... I'd add these in the `Noir` API, together with the other init functions `initACVM` and `initAbi`

this will make the app have a bit of an overhead at the start, but from then on every proof will _apparently_ be much faster

Reply:
In other places we do lazy instantiation, I have added an issue here on either doing one way or the otherhttps://github.com/noir-lang/noir/issues/2910#issuecomment-1742811237